### DISCUSSION TITLE
segmentation error while typing "forge init"

### DISCUSSION THREAD

**Discussion by 0xkumar** [Cyfrin/foundry-full-course-cu#1190] [Q&A]
I tried to create a new project so open created an new empty directory and in the vs code i tried the command "forge init" but it is always giving me "segmentation fault". The chatgpt answer is "It's a mechanism that prevents you from corrupting the memory and introducing hard-to-debug memory bugs. Whenever you get a segmentation fault, you know you are doing something wrong with memory - such as accessing a variable that has already been freed or writing to a read-only portion of memory".Anyone have the idea about it

**Comment by xyz899** [comment]
Hi,

That's very weird considering you're only using the command-line tool. 
The Forge binary or some of its dependencies might be corrupted. Try reinstalling Forge. You can uninstall it and then install it again `curl -L https://foundry.paradigm.xyz | bash` then run `foundryup` to see if that resolves the issue. Be aware that if you are on windows Foundry cannot be installed within powershell you need WSL.

**Reply by 0xkumar** [reply]
Hi,
This might be a noob question.How can i unistall foudndry in wsl?

**Reply by xyz899** [reply]
Find all binaries paths ( cast, anvil, forge, etc..) use`rm /path/to/anvil` for example one by one

**Reply by 0xkumar** [reply]
I tried foundryup and the issue was rectified, Thank you.
